device/src/main/java/com/softserve/lv460/device/config/ChangeStreamConfig.java

```diff
@@ -0,0 +1,123 @@
+package com.softserve.lv460.device.config;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.mongodb.Block;
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import com.softserve.lv460.device.action.ActionExecutor;
+import com.softserve.lv460.device.controller.DeviceDataController;
+import com.softserve.lv460.device.dto.rule.ActionRule;
+import com.softserve.lv460.device.dto.rule.RuleDto;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+@Service
+public class ChangeStreamConfig {
+  private final MongoClient mongoClient;
+  private final CloseableHttpClient httpClient;
+  private final PropertiesConfig propertiesConfig;
+  private final ActionExecutor actionExecutor;
+  private LoadingCache<String, List<RuleDto>> cache;
+  private static Logger logger = LoggerFactory.getLogger(DeviceDataController.class);
+
+  public ChangeStreamConfig(MongoClient mongoClient, CloseableHttpClient httpClient, PropertiesConfig propertiesConfig, ActionExecutor actionExecutor) {
+    this.httpClient = httpClient;
+    this.propertiesConfig = propertiesConfig;
+    this.actionExecutor = actionExecutor;
+    this.mongoClient = mongoClient;
+    this.cache = CacheBuilder
+            .newBuilder()
+            .expireAfterWrite(propertiesConfig.getCacheExpiration(), TimeUnit.MINUTES)
+            .build(
+                    new CacheLoader<String, List<RuleDto>>() {
+                      @Override
+                      public List<RuleDto> load(String uuId) throws IOException {
+                        return getRules(uuId);
+                      }
+                    }
+            );
+  }
+
+  @PostConstruct
+  public void subscribeToChangeStream() {
+    Thread thread = new Thread(() -> {
```

Comment:
1. how do you handle thread termination when application stops?
2. It's better to use ExecutorService